pylot/plugins/helpers/pylot_helpers.py

```diff
@@ -18,34 +18,22 @@ def get_config_options(cls):
             options = json.load(_file)
         return options
 
-    @staticmethod
-    def get_hash_token_file():
-        now: datetime = datetime.now()
-        captured_time: int = 60
-        for ele in range(0, 60, 15):
-            if now.minute < ele:
-                captured_time = ele
-                break
-        return f"{now.day}-{now.hour}-{captured_time}"
-
     @classmethod
     def get_cumulus_api_instance(cls):
-        """ Get Cumulus instance with cached token"""
-        if 'PRIVATE_API_LAMBDA_ARN' in os.environ:
-            print('getting non-token instance')
-            cml = CumulusApi()
-            pass
-        else:
-            print('getting token instance')
-            get_hashed_file_name = cls.get_hash_token_file()
-            token: str
-            tempfile = f'{mkdtemp()}/{get_hashed_file_name}'
-            if not os.path.isfile(tempfile):
-                with open(tempfile, 'w', encoding='utf-8') as _file:
-                    cml = CumulusApi()
-                    _file.write(cml.TOKEN)
-            else:
-                with open(tempfile, 'r', encoding='utf-8') as _file:
+        token_dir = f'{gettempdir()}/pylot_token/'
+        os.makedirs(token_dir, exist_ok=True)
+        token_file = f'{token_dir}token'
+        token = None
+        if os.path.isfile(token_file):
+            print(f'Using local token: {token_file}')
+            file_stat = os.stat(token_file)
+            if int(time()) - int(file_stat.st_mtime) < 3600:
```

Comment:
Good use of the os.stat library for determining whether or not the token has expired. 👍 I imagined this was going to be tough to account for, but I think this solution will work great for our general use case.